src/ds323x/alarms.rs

```diff
@@ -113,20 +114,46 @@ where
         when: DayAlarm1,
         matching: Alarm1Matching,
     ) -> Result<(), Error<CommE, PinE>> {
-        if when.day < 1 || when.day > 31 || when.minute > 59 || when.second > 59 {
+        let day_invalid = when.day < 1 || when.day > 31;
+        let seconds_invalid = when.second > 59;
+        let minutes_invalid = when.minute > 59;
+
+        let day = if day_invalid { 1 } else { when.day };
+        let minute = if minutes_invalid { 0 } else { when.minute };
+
+        if (matching == Alarm1Matching::AllMatch && day_invalid)
+            || ((matching != Alarm1Matching::SecondsMatch
+                && matching != Alarm1Matching::OncePerSecond)
+                && minutes_invalid)
+            || seconds_invalid
+        {
             return Err(Error::InvalidInputData);
         }
+
         let match_mask = get_matching_mask_alarm1(matching);
         let mut data = [
             Register::ALARM1_SECONDS,
             decimal_to_packed_bcd(when.second) | match_mask[0],
-            decimal_to_packed_bcd(when.minute) | match_mask[1],
+            decimal_to_packed_bcd(minute) | match_mask[1],
             hours_to_register(when.hour)? | match_mask[2],
-            decimal_to_packed_bcd(when.day) | match_mask[3],
+            decimal_to_packed_bcd(day) | match_mask[3],
         ];
         self.iface.write_data(&mut data)
     }
 
+    /// Set Alarm1 for a time (fires when hours, minutes and seconds match).
+    ///
+    /// Will return an `Error::InvalidInputData` if any of the parameters is out of range.
+    pub fn set_alarm1_hms(&mut self, when: NaiveTime) -> Result<(), Error<CommE, PinE>> {
+        let alarm = DayAlarm1 {
+            day: 0,
+            hour: Hours::H24(when.hour() as u8),
+            minute: when.minute() as u8,
+            second: when.second() as u8,
+        };
+        self.set_alarm1_day(alarm, Alarm1Matching::HoursMinutesAndSecondsMatch)
+    }
+
     /// Set Alarm1 for weekday.
     ///
     /// Will return an `Error::InvalidInputData` if any of the parameters is out of range.
```

Comment:
Same as above: Could you document here how the values of invalid inputs are adapted when these are ignored by the selected matching strategy?